usecases/auth/authorization/rbac/authorizer.go

```diff
@@ -77,3 +90,81 @@ func (m *manager) Authorize(principal *models.Principal, verb string, resources
 func (m *manager) AuthorizeSilent(principal *models.Principal, verb string, resources ...string) error {
 	return m.authorize(principal, verb, true, resources...)
 }
+
+func (m *manager) FilterAuthorizedResources(principal *models.Principal, verb string, resources ...string) ([]string, error) {
+	if principal == nil {
+		return nil, errors.NewUnauthenticated()
+	}
+
+	if len(resources) == 0 {
+		return nil, fmt.Errorf("at least 1 resource is required")
+	}
+
+	logger := m.logger.WithFields(logrus.Fields{
+		"action":         "authorize",
+		"user":           principal.Username,
+		"component":      authorization.ComponentName,
+		"request_action": verb,
+	})
+	if len(principal.Groups) > 0 {
+		logger = logger.WithField("groups", principal.Groups)
+	}
+
+	var permResults []logrus.Fields
+	var allowedResources []string
+
+	// First try with wildcard
+	parts := strings.Split(resources[0], "/")
+	parts[len(parts)-1] = "*"
+	wildcardPath := strings.Join(parts, "/")
+
+	allowed, err := m.checkPermissions(principal, wildcardPath, verb)
+	if err != nil {
+		logger.WithError(err).WithField("resource", wildcardPath).Error("failed to enforce policy")
+		return nil, err
+	}
+
+	if allowed {
+		// If wildcard is allowed, log permissions for all requested resources
+		for _, resource := range resources {
+			perm, err := conv.PathToPermission(verb, resource)
+			if err != nil {
+				return nil, err
+			}
+
+			permResults = append(permResults, logrus.Fields{
+				"resource": prettyPermissionsResources(perm),
+				"results":  prettyStatus(true),
+			})
+		}
+
+		logger.WithField("permissions", permResults).Info()
+		return resources, nil
+	}
+
+	// If wildcard check failed, check individual resources
+	for _, resource := range resources {
+		allowed, err := m.checkPermissions(principal, resource, verb)
+		if err != nil {
+			logger.WithError(err).WithField("resource", resource).Error("failed to enforce policy")
+			return nil, err
+		}
+
+		perm, err := conv.PathToPermission(verb, resource)
+		if err != nil {
+			return nil, err
+		}
+
+		permResults = append(permResults, logrus.Fields{
+			"resource": prettyPermissionsResources(perm),
+			"results":  prettyStatus(allowed),
+		})
```

Comment:
I am not convinced that we need to add each resource to the log. I think it would make more sense if we would do "User requested to list all collections, checking which ones a user has access to" or something with the same meaning.

Reply:
from audit log PoV we shall log every resource the user had access to , this case we will differentiate between had access to all or scooped view of the resources

Reply:
> from audit log PoV we shall log every resource the user had access to

but if there are 50k tenants a single tenants_get() will produce 50k log lines. The user is not trying to access anything forbidden but is just saying "please give me all tenants I have access to"

Reply:
> but if there are 50k tenants a single tenants_get() will produce 50k log lines
yes this is audit logging 
but anyway this line is not line per resource, it's field per resource

Reply:
> yes this is audit logging

This feels very unreasonable to me. With every request you get a list of all your tenants/collections and if the user has access to them or not. I don't see the value of this?

For me there is a fundamental difference between RBAC as gatekeeper (user wants X, RBAC says yes/no) and RBAC as filter (user wants to get what he has permission to).

> anyway this line is not line per resource, it's field per resource

oki true, but doesn't change the underlying argument of the log becoming gigantic

Reply:
(posted this in the wrong thread first, deleted there)

I don't see how this is needed for compliance - maybe we should take this offline and/or involve more people, but I am currently not convinced this is needed. Happy to be wrong about this, but I currently cannot see why anyone would want this behaviour.

> this is a seperate and internal general issue, if we paginate our requests the log won't be huge, we shouldn't allow requesting 50k in 1st place

if you paginate through everything, the log will still be gigantc, just spread out to different calls

Reply:
> I don't see how this is needed for compliance - maybe we should take this offline and/or involve more people, but I am currently not convinced this is needed

that is the definition of audit logging! see https://audit-logs.tax, if the amount of resources are huge this is not audit log issue but the end point issue to allow that much of data requested in 1 call 

let's bring it offline